docs/user/testing.md

```diff
@@ -0,0 +1,182 @@
+
+Let's write a simple test case for the vpg algorithm `src/garage/tf/algos/vpg.py`.
+
+We begin by creating a file called `test_vpg.py` and put it under the tree `tests/garage/tf/algos/`.
+
+We want to test VPG in the cart pole environment. Hence, we create a unitest named `test_vpg_cartpole`. Inside the function, we define the environemnt, the policy and the baselien and feed them to the VPG algorithm and run the experiment. Finally, we make sure the return value is identical to our expectation by using `assert`.
+
+In new xunit-style tests, multiple tests are modeled into a class for modular and scalable structure. There is no need to subclass or anything, but make sure the prefix of the class starts with `Test`, otherwise the class will be skipped. Note that it's not encouraged to use this style, especially when a test doesn't require `setup_method` and `teardown_method`.
+
+```python
+# test_vpg.py
+
+class TestVPG(...):
+    def test_vpg_cartpole(self):
+        ...
+        env = GarageEnv(env_name='CartPole-v1')
+
+            policy = CategoricalMLPPolicy(name='policy',
+                                          env_spec=env.spec,
+                                          ...)
+
+            baseline = LinearFeatureBaseline(env_spec=env.spec)
+
+            algo = VPG(env_spec=env.spec,
+                       policy=policy,
+                       baseline=baseline,
+                       max_path_length=100,
+                       discount=0.99,
+                       optimizer_args=dict(learning_rate=0.01, ))
+
+            runner.setup(algo, env)
+
+            last_avg_ret = runner.train(n_epochs=10, batch_size=10000)
+            assert last_avg_ret > 90
+
+            env.close()
+```
+
+## Run tests
+
+In general, we can start running tests simply by:
+
+```bash
+$ pytest
+```
+
+However, in most of the cases, we simply don't have the time to test everything, plus Travis CI will take care of the  majority of tests upon the deployment workflow. We can use the following ways to carry out specific tests to make life easier.
+
+### Specifying tests / selecting tests
+
+Pytest supports several ways to run and select tests.
+
+#### Run tests in a directory
+
+Run a test on particular module(s) by specifying a directory path.
+
+```bash
+$ pytest tests/garage/tf/algo/
+```
+
+#### Run tests in a module
+
+Run a test on a particular module by specifying a file path.
+
+```bash
+$ pytest tests/garage/tf/algo/test_ppo.py
+```
+
+#### Run tests by keyword expressions
+
+Run tests by keyword expressions. This is useful for running particular test function(s).
+
+```bash
+$ pytest -k test_ppo_pendulum_continuous_baseline
+
+```
+
+## Useful garage methods
+
+Garage uses two helper functions `setup_method` and `teardown_method` to set up and close down test environment properly. For details, check [this](https://docs.pytest.org/en/2.8.7/xunit_setup.html).
```

Comment:
this is actually just a feature of pytest (xunit support), but here i think you imply garage introduced this specifically.